index.js

```diff
@@ -145,6 +170,17 @@ function watchify (b, opts) {
 
     b.on('transform', function (tr, mfile) {
         tr.on('file', function (dep) {
+            cache._transformDeps[mfile] = cache._transformDeps[mfile] || [];
+            cache._transformDeps[mfile].push(dep);
```

Comment:
Ah, good catch!  I think it's actually more correct to `delete cache._transformDeps[mfile]` on `mfile transform`.

Another test case might be changing the dependencies, then modifying an old dependency and confirming a recompile isn't triggered.